### DISCUSSION TITLE
PostProcessor "ElementAverageMaterialProperty" applied to "ADParsedMaterial"?

### DISCUSSION THREAD

**Discussion by EdSheltonUKAEA** [idaholab/moose#22955] [Q&A Modules: General] [answered]
I would like to use the postprocessor "ElementAverageMaterialProperty".  But I am using AD kernels, boundary conditions, material properties, etc. including "ADParsedMaterial" to define the parameters I want to track in the PP.

When I try to run an input file including the "ElementAverageMaterialProperty" PP, I get this error message. 
```
*** ERROR ***
The requested regular material property Focc_tr_def1 is declared as an AD property. Either retrieve it as an AD property with getADMaterialProperty or declare it as a regular property with declareProperty
```

What changes do I need to make to my input files, and in which blocks, to do either or both of these suggested fixes?

Or is there an equivalent PP which will work directly with the type "ADParsedMaterial"?

Many thanks!

**Comment by maxnezdyur** [comment, marked as answer]
Try `ADElementAverageMaterialProperty`.

**Reply by EdSheltonUKAEA** [reply]
Thanks!  I thought there must be something like this, but couldn't see it in the list here:
https://mooseframework.inl.gov/source/index.html 

This has worked!

**Reply by GiudGiud** [reply]
Sorry we don’t double up the documentation entries for the AD versions.
